### PR TITLE
move some sparse docs from helpdb.jl

### PR DESCRIPTION
**Pull request by KristofferC**
I get a bunch of unrelated diffs when I run `genstdlib.jl` so it seems that the `.rst` files and the julia docs are not in sync atm. 

Maybe someone could commit a `genstdlib.jl` run to master and I could rebase upon that?

### CODE REVIEW

File: base/sparse/sparsematrix.jl [hunk 1]
```diff
@@ -2565,6 +2678,15 @@ function spdiagm_internal(B, d)
     return (I,J,V)
 end
 
+doc"""
```
Comment (MichaelHatherly) [inline base/sparse/sparsematrix.jl]: Doesn't seem to need `doc"` prefix here. Probably best to keep things consistent and only use `doc"` when necessary.

Comment (KristofferC) [inline base/sparse/sparsematrix.jl]: yeah, I just missed it. Fixing

Comment (KristofferC) [inline base/sparse/sparsematrix.jl]: done

**Comment by tkelman** [comment]
I don't see any diffs when i run genstdlib on master. Be sure you rebuild julia and use that just-built julia to run genstdlib

**Comment by KristofferC** [comment]
That might be it. I forgot to rebuild julia. Gimme a sec and I'll check.

**Comment by KristofferC** [comment]
Hmm,

Julia updated and built on master:

image

After `genstdlib.jl`

image

**Comment by tkelman** [comment]
Could you gist the diff?

**Comment by KristofferC** [comment]
https://gist.github.com/KristofferC/8c2be24cb47c189cc5ec

**Comment by tkelman** [comment]
Could you force a re-build with a `make clean` first? The change at the very end where `The default handler returns the exception if it is of the expected type.` is getting deleted is reversing a recent change, so I suspect your system image still might be out of date?

**Comment by KristofferC** [comment]
I'm on commit 31f618142, did `make clean` and `make` ran `genstdlib.jl` and got the same diff as in the gist. Weird...

**Comment by KristofferC** [comment]
My `helpdb.jl` contains the correct doc string for `@test_throws`. Before `genstdlib.jl` is run the doc string in the `.rst` file and `helpdb.jl` are in agreement. Then after I run `genstdlib.jl` it adds the "The default handler returns the exception if it is of the expected type."

I have no idea where it finds that string though...

**Comment by KristofferC** [comment]
I got it to work now without getting spurious changes after `genstdlib.jl`. There are some changes for one of the docs in the generated .rst file but it looks like whitespace only.

File: base/sparse/sparsematrix.jl [hunk 2]
```diff
@@ -25,11 +25,46 @@ function SparseMatrixCSC(m::Integer, n::Integer, colptr::Vector, rowval::Vector,
...
 rowvals(S::SparseMatrixCSC) = S.rowval
+
+"""
+    nzrange(A, col)
+
+Return the range of indices to the structural nonzero values of a sparse matrix
+column. In conjunction with `nonzeros(A)` and `rowvals(A)`, this allows for
+convenient iterating over a sparse matrix :
+
+    A = sparse(I,J,V)
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: should this be fenced with markdown syntax \`\`\`jl ?

File: base/sparse/sparsematrix.jl [hunk 3]
```diff
@@ -2565,6 +2678,15 @@ function spdiagm_internal(B, d)
     return (I,J,V)
 end
 
+"""
+    spdiagm(B, d[, m, n])
+
+Construct a sparse diagonal matrix. `B` is a tuple of vectors containing the diagonals and
+`d` is a tuple containing the positions of the diagonals. In the case the input contains only
+one diagonaly, `B` can be a vector (instead of a tuple) and `d` can be the diagonal position
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: may as well fix the typo, s/diagonaly/diagonal/

File: base/sparse/sparsematrix.jl (hunk 2 above)
Comment (KristofferC) [inline base/sparse/sparsematrix.jl]: I thought indented text defaulted to code boxes and the default language we have set is Julia so specifying `jl` doesn't do anything. I can of course change it if you still want.

File: base/sparse/sparsematrix.jl [hunk 4]
```diff
@@ -212,9 +247,20 @@ function vec{Tv,Ti}(S::SparseMatrixCSC{Tv,Ti})
     SparseMatrixCSC(lS, 1, colptr, rowval, copy(S.nzval))
 end
 
+"""
+    sparse(A)
+
+Convert an AbstractMatrix `A` into a sparse matrix.
+"""
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: this is on the wrong method

File: base/sparse/sparsematrix.jl (hunk 2 above)
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: you're right, nevermind - it's already coming through as a code block in the rst, fine as is

**Comment by KristofferC** [comment]
Should I amend the previous commit or commit a new one with the fixes to the comments? Easier to review with a new commit but then I have to squash them later.

File: base/sparse/sparsematrix.jl [hunk 5]
```diff
@@ -458,18 +520,57 @@ function sprand{T}(m::Integer, n::Integer, density::AbstractFloat,
...
+..  sprand([rng,] m,n,p [,rfn])
+
+Create a random ``m`` by ``n`` sparse matrix, in which the probability of any
+element being nonzero is independently given by ``p`` (and hence the mean
+density of nonzeros is also exactly ``p``). Nonzero values are sampled from
+the distribution specified by ``rfn``. The uniform distribution is used in
+case ``rfn`` is not specified. The optional ``rng`` argument specifies a
+random number generator, see :ref:`Random Numbers <random-numbers>`.
+```
+"""
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: would it make more sense to put this on the method that has both `rng` and `rfn` arguments?

**Comment by tkelman** [comment]
new commit makes it easier to track incremental changes, squashing in the end isn't absolutely required unless there are a large number of review commits or some of the intermediate commits would fail tests

**Comment by KristofferC** [comment]
New commit added with review comments addressed.

**Comment by KristofferC** [comment]
Should be ok to merge if no one else has comments.

**Comment by jakebolewski** [comment]
lgtm

**Comment by KristofferC** [comment]
Worth to backport?

**Comment by tkelman** [comment]
may as well, shouldn't hurt anything

**Pull request merged**